data/mappings/defaults.hjson

```diff
@@ -8,43 +8,37 @@
         "elite_c": {
             "processor": "atmega32u4",
             "bootloader": "atmel-dfu",
-            "pin_compatible": "promicro"
+            "pin_compatible": "elite_c"
         },
```

Comment:
This is technically correct, but would change the behavior for some existing boards that have `"development_board": "elite_c"`:
```
keyboards/argo_works/ishi/80/mk0_avr_extra/info.json:    "development_board": "elite_c",
keyboards/boardsource/lulu/avr/info.json:  "development_board": "elite_c",
keyboards/handwired/onekey/elite_c/info.json:    "development_board": "elite_c",
keyboards/splitkb/aurora/corne/info.json:    "development_board": "elite_c",
keyboards/splitkb/aurora/lily58/info.json:    "development_board": "elite_c",
keyboards/splitkb/aurora/sweep/info.json:    "development_board": "elite_c",
```
- `argo_works/ishi/80/mk0_avr_extra` actually uses the extra Elite-C pins, and this fix would actually allow removing the `"pin_compatible": "elite_c"` override from that board (and another `PIN_COMPATIBLE = elite_c` override in `rules.mk` which was added as a workaround for a bug that's getting fixed in #19511).
- `boardsource/lulu/avr` uses pins `F0` and `F1`, so `"pin_compatible": "elite_c"` would be correct for that board.
- `handwired/onekey/elite_c` is obviously Elite-C.
- `splitkb/aurora/corne`, `splitkb/aurora/lily58` and `splitkb/aurora/sweep` apparently use only the Pro Micro pins (the PCBs don't even have holes for extra Elite-C pins), so these boards would need to have `"pin_compatible": "promicro"` added (and #19511 to make that actually work).

Reply:
Also if this fix goes into master without #19511, the mentioned splitkb boards would need to have `PIN_COMPATIBLE = promicro` added to `rules.mk` instead.

Reply:
Ah good point, lets keep it as is for now.